packages/timeline/src/ReactGantt.ts

```diff
@@ -0,0 +1,467 @@
+                                _id: d.id,
+                                element: () => d.element
+                            }, 
+                            d3Event
+                        );
+                        context.selectionChanged();
+                    })
+                    .on("click", function (this: SVGElement, d) {
+                        const selected = d.element.classed("selected");
+                        context.item_click(d.props.origData || d.props, "", selected);
+                    })
+                    .on("mousein", function (d) {
+                        context.highlightItem(d3Select(this), d);
+                        const selected = d.element.classed("selected");
+                        context.item_mousein(d.props.origData || d.props, "", selected);
+                    })
+                    .on("mouseover", function (d) {
+                        const d3evt = d3Event();
+                        context._tooltip._triggerElement = d.element;
+                        context._tooltip._cursorLoc = [
+                            d3evt.clientX,
+                            d3evt.clientY
+                        ];
+                        context._tooltip
+                            .data(d)
+                            .visible(true)
+                            .render()
+                            ;
+                        context.highlightItem(d3Select(this), d);
+                        const selected = d.element.classed("selected");
+                        context.item_mouseover(d.props.origData || d.props, "", selected);
+                    })
+                    .on("mouseout", function (d) {
+                        context._tooltip
+                            .visible(false)
+                            .render()
+                            ;
+                        context.highlightItem(null, null);
+                        const selected = d.element.classed("selected");
+                        context.item_mouseout(d.props.origData || d.props, "", selected);
+                    })
+                    .each(function (d, i) {
+                        d.element = d3Select(this);
+                        d.x = context._interpolateX(d[1]);
+                        const endX = context._interpolateX(d[2]);
+                        d.y = context._interpolateY(context._buckets[i]);
+                        d.props={
+                            ...d[3],
+                            text: d[0]
+                        };
+                        d.props.width = endX - d.x;
+                        d.props.height = context._bucketHeight;
+                        d.x += borderOffset1;
+                        d.y += borderOffset1;
+                        d.props.width -= borderOffset2;
+                        d.props.height -= borderOffset2;
+                        d.element.attr("transform", `translate(${d.x+(d.props.width/2)} ${d.y+(d.props.height/2)})`);
+                    }),
```

Comment:
The text and end time etc. could change over time.

Reply:
Yes, this just sets the initial transform

Reply:
If you set it in the update as well then this is not needed?

Reply:
@jbrundage bump

Reply:
We discussed this on the phone the other day and determined that this line was acceptable.